stdlib/typing.pyi

```diff
@@ -631,6 +631,7 @@ class Mapping(Collection[_KT], Generic[_KT, _VT_co]):
     def keys(self) -> KeysView[_KT]: ...
     def values(self) -> ValuesView[_VT_co]: ...
     def __contains__(self, __key: object) -> bool: ...
+    def __eq__(self, other: object) -> bool: ...
```

Comment:
I think mypy requires this to be `object -> bool`. I'm guessing it doesn't have any logic to deal with `Notimplemented`?
Otherwise, this should actually be `Mapping[Any, Any] -> bool` and `object -> NotImplemented`.

Reply:
Type checkers don't understand `NotImplemented`. The way you've typed it is correct.

Reply:
That's fine. I guess if they did understand it though, then they would be able to give an error when you try to compare Mapping to another object with `__eq__()`, realising that they can't be compared together (although in practice, I struggle to find an object that doesn't compare...).

Reply:
The heuristics mypy uses for its overlapping-equality check turn out to be pretty good in practice, so I don't think that's a huge issue. Though I agree it can be somewhat surprising when writing stubs that adding `__eq__` methods with the exact same signature to subclasses can have an impact on typechecking

Reply:
(Also, to note, if type checkers _did_ understand `NotImplemented` it would be _very_ tedious for us, as _every_ BinOp method would have to be typed using overloads 😄)